### ISSUE TITLE
Player dosn't work with audi

### ISSUE THREAD

**Issue by CuBa94** [melnijir/Dicaffeine#2]

**Comment by CuBa94** [comment]
Hello,

Yesterday I installed the latest version of Dicaffeine on my Raspberry 4. For this I first did a clean install of Raspberry Pi OS with desktop (Debian Bullseye) and updated everything.

Now the player does not work for me when I select the setting "Internal Audio Analog Stereo" or "Internal Audio Digital Stereo". The NDI stream comes from ProPresenter 7 with the latest version 7.8.

What is the problem here?

-> A video of the error: https://1drv.ms/v/s!AtXbThOljQDzyxSkeMUMbB53HxUo?e=7EbqXt

**Comment by melnijir** [comment]
Hi,

can you start the stream as showed in the video (when it does not start) and then open the terminal window and send me the output of command:

`yuri2 /tmp/yuri_config_player.xml`

Thanks!

**Comment by CuBa94** [comment]
Do you mean this?

pi@raspberrypi:~ $ yuri2 /tmp/yuri_config_player.xml
WARNING 29: 12:53:51 [NDIInput/ndi_input_0] Error setting parameter event_time
INFO 35: 12:53:51 [webserver/websrv] Created socket
INFO 37: 12:53:51 [web_static/webstatic] Resizing to 12
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Geometry 800x600+0+0
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Connected to display :0.0
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Screen number is 0
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Found root window
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Found visual 614
INFO 39: 12:53:51 [glx_window/window_HDMI-1] geometry 1920x1080+0+0
INFO 39: 12:53:51 [glx_window/window_HDMI-1] X Window Created
INFO 35: 12:53:51 [webserver/websrv] Starting worker thread
INFO 37: 12:53:51 [web_static/webstatic] Registered to server
INFO 35: 12:53:51 [webserver/websrv] Helper thread started
INFO 33: 12:53:51 [web_control/webctrl] Registered to server
INFO 29: 12:53:51 [NDIInput/ndi_input_0] Found stream "MACBOOK-PRO-VON-CURDIN.HOME (MacBook Pro von Curdin - NDI 1)" with id 0
INFO 29: 12:53:51 [NDIInput/ndi_input_0] Receiving started
INFO 29: 12:53:51 [NDIInput/ndi_input_0] Sender doest not support PTZ, disabling events.
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Created GLX Context
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Decorations XChangeProperty returned 1
INFO 39: 12:53:51 [glx_window/window_HDMI-1] setting on top: 0
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Fullscreen XChangeProperty returned 1
INFO 39: 12:53:51 [glx_window/window_HDMI-1] Stereo method: none
INFO 41: 12:53:51 [GL] Generated texture 1
ERROR 41: 12:53:51 [GL] Error 1280 setting texture params
ERROR 41: 12:53:51 [GL] Error 1280 setting texture params
INFO 18: 12:53:52 [pulse_output/audio_output] Initialized for 48000 Hz
INFO 18: 12:53:52 [pulse_output/audio_output] Initialized for 8 channels
ERROR 18: 12:53:52 [pulse_output/audio_output] Error while creating new pulse audio playback.
Assertion 's' failed at pulse/stream.c:1483, function pa_stream_write_ext_free(). Aborting.
Abgebrochen

**Comment by melnijir** [comment]
Yes, thanks - I can see one thing, the audio has 8 channels and the player does not know how to map them to stereo sound. That should be fixed, thanks for reporting. For now - can you limit the sender to only 2 audio channels and check if it helps?

**Comment by CuBa94** [comment]
> 

Sorry, no i can't limit the sender to only 2 audio channels... ProPresenter sends automaticlly 8 channels...

**Comment by melnijir** [comment]
That's unfortunate, NDI receiver in Dicaffeine detects 8 channels and tries to configure the audio output for it. I'll fix that in the next release - only maximal available channels should be used.

**Comment by CuBa94** [comment]
> That's unfortunate, NDI receiver in Dicaffeine detects 8 channels and tries to configure the audio output for it. I'll fix that in the next release - only maximal available channels should be used.

Yes, you are right. Unfortunately, the number of channels cannot be edited in ProPresenter. If you want to test your fix, ProPresenter can be used for free with watermark. You can easily set up an NDI stream and set audio for SDI & NDI in the audio settings.

ProPresenter can be downloaded here: https://renewedvision.com/propresenter/#download

Thanks for your work! Am big fan of Dicaffeine! :-)

**Comment by melnijir** [comment]
Thanks :-)

I haven't try it yet (seems that ProPresenter is for Mac/Windows and I have only Linux here), but if you want, you can test the new version:

https://dicaffeine.com/nightbuilds

I've added small fix checking the maximal channels count. It's tested only with virtual sound generator, so I am not really sure if it will work.
